dev/com.ibm.ws.security.wim.adapter.ldap/src/com/ibm/ws/security/wim/adapter/ldap/LdapConnection.java

```diff
@@ -2143,14 +2163,41 @@ private void supportRangeAttributes(Attributes attributes, String dn, TimedDirCo
         }
     }
 
+    /**
+     * Find the range returned from LDAP. This will be the maximum LDAP will return so
+     * we want to use this rather than the default attributeRangeStep.
+     *
+     * @param attr The range attribute returned from LDAP, eg. member;range=0-1499
+     * @return The max value of results LDAP will return
+     */
+    private int parseRangeAttribute(String attr) {
+        try {
+            //Find the range by parsing member;range=1500-2999 for low and high.
+            int rangeLow = Integer.parseInt(attr.substring(attr.indexOf('=') + 1, attr.indexOf('-')));
+            int rangeHigh = Integer.parseInt(attr.substring(attr.indexOf('-') + 1));
```

Comment:
What happens when the range is `1000-*`? This would fail with a NumberFormatException.

Reply:
Or do we not have tow worry about that here? Is this only called after the initial call where we won't get a range with '*' in it?

Reply:
Right, this is only called once after the initial call. The initial call won't return * because if the number of attributes is that low, there doesn't need to be a range.